opentelemetry-api/src/opentelemetry/context/__init__.py

```diff
@@ -138,15 +138,34 @@ def __repr__(self):
         asyncio.run(main())
 """
 
-from .base_context import BaseRuntimeContext
+import typing
+from contextlib import contextmanager
 
-__all__ = ["Context"]
+from .base_context import Context
 
-try:
-    from .async_context import AsyncRuntimeContext
 
-    Context = AsyncRuntimeContext()  # type: BaseRuntimeContext
-except ImportError:
-    from .thread_local_context import ThreadLocalRuntimeContext
+def current() -> Context:
+    return _CONTEXT.current()
 
-    Context = ThreadLocalRuntimeContext()
+
+def new_context() -> Context:
+    try:
+        from .async_context import (  # pylint: disable=import-outside-toplevel
+            AsyncRuntimeContext,
+        )
+
+        context = AsyncRuntimeContext()  # type: Context
+    except ImportError:
+        from .thread_local_context import (  # pylint: disable=import-outside-toplevel
+            ThreadLocalRuntimeContext,
+        )
+
+        context = ThreadLocalRuntimeContext()  # type: Context
+    return context
+
+
+def merge_context_correlation(source: Context, dest: Context) -> Context:
+    return dest.merge(source)
+
+
+_CONTEXT = new_context()
```

Comment:
Probably you don't need to have an object here but the **type** of context to use.